opencti-platform/opencti-front/src/components/ItemOpenVocab.tsx

```diff
@@ -1,33 +1,57 @@
 import React, { FunctionComponent } from 'react';
 import makeStyles from '@mui/styles/makeStyles';
-import * as R from 'ramda';
+import Chip from '@mui/material/Chip';
 import Tooltip from '@mui/material/Tooltip';
 import { InformationOutline } from 'mdi-material-ui';
 import { graphql, usePreloadedQuery } from 'react-relay';
 import { PreloadedQuery } from 'react-relay/relay-hooks/EntryPointTypes';
 import { useFormatter } from './i18n';
+import { Theme } from './Theme';
 import useVocabularyCategory from '../utils/hooks/useVocabularyCategory';
 import { ItemOpenVocabQuery } from './__generated__/ItemOpenVocabQuery.graphql';
 import useQueryLoading from '../utils/hooks/useQueryLoading';
 
-const useStyles = makeStyles(() => ({
+const useStyles = makeStyles<Theme>((theme) => ({
   container: {
     margin: 0,
     padding: 0,
     display: 'flex',
   },
+  chip: {
+    fontSize: 12,
+    marginRight: 7,
+    borderRadius: '0',
+    width: 120,
+    height: 'auto',
+    color: theme.palette.mode === 'dark' ? '#ffffff' : '#000000',
+    borderColor: theme.palette.mode === 'dark' ? '#ffffff' : '#000000',
+    backgroundColor: theme.palette.mode === 'dark' ? 'rgba(255, 255, 255, .1)' : 'rgba(0, 0, 0, .1)',
+    '& .MuiChip-label': {
+      whiteSpace: 'normal',
+      padding: '4px 6px',
+    },
+  },
   icon: {
     margin: '15px 0 0 10px',
   },
   smallIcon: {
     margin: '5px 0 0 10px',
   },
+  pre: {
+    marginTop: '7px',
+  },
+  smallPre: {
+    margin: 0,
+    paddingTop: '7px',
+    paddingBottom: '4px',
+  },
 }));
 
 interface ItemOpenVocabProps {
   type: string;
   value?: string | null;
-  small: boolean;
+  small?: boolean;
+  chipDisplay?: boolean;
```

Comment:
Could be better to have an enum for display mode ? What do you think ?
type displayMode = 'chip' | 'span'

Reply:
done